src/datasets/data_files.py

```diff
@@ -544,8 +544,8 @@ def _get_single_origin_metadata(
 
 def _get_origin_metadata(
     data_files: List[str],
-    max_workers=64,
     download_config: Optional[DownloadConfig] = None,
+    max_workers: int = 16,
```

Comment:
Maybe this should be a config variable (which we would also use in `DownloadManager`)